scripts/zephyr_module.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2019, Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Tool for parsing a list of projects to determine if they are Zephyr
+projects. If no projects are given then the output from `west list` will be
+used as project list.
+
+Include file is generated for Kconfig using --kconfig-out.
+A <name>:<path> text file is generated for use with CMake using --cmake-out.
+'''
+
+import argparse
+import os
+import sys
+import yaml
+import pykwalify.core
+import subprocess
+import re
+
+
+METADATA_SCHEMA = '''\
+## A pykwalify schema for basic validation of the structure of a
+## metadata YAML file.
+##
+# The zephyr/module.yml file is a simple list of key value pairs to be used by
+# the build system.
+type: map
+mapping:
+  build:
+    required: true
+    type: map
+    mapping:
+      cmake:
+        required: false
+        type: str
+      kconfig:
+        required: false
+        type: str
+'''
+
+AUTO_GENERATED_HEADER = '# NOTE: THIS FILE IS AUTOGENERATED - DO NOT EDIT\n\n'
+
+
+def process_module(module, section_name, cmake_out=None, kconfig_out=None):
+    cmake_path = 'zephyr'
+    kconfig_file = 'zephyr/Kconfig'
+
+    module_yml = os.path.join(module, 'zephyr/module.yml')
+    if os.path.isfile(module_yml):
+        with open(module_yml, 'r') as f:
+            meta = yaml.safe_load(f.read())
+
+        schema = yaml.safe_load(METADATA_SCHEMA)
+        pykwalify.core.Core(source_data=meta, schema_data=schema).validate()
+
+        section = meta.get(section_name, dict())
+        if section is not None:
+            cmake_path = section.get('cmake', cmake_path)
+            kconfig_file = section.get('kconfig', kconfig_file)
+
```

Comment:
we should assert when the yaml file is pointing to a non-existent file.

Reply:
I think you're referring to using try statements to open the file, and doing something like calling `sys.exit` or printing to stderr in the exception handler, right? Not an actual `assert` statement.

Reply:
@SebastianBoe  yep, see your point, although I agree with @mbolivar  you don't want assert, but instead raise an exception in order to abort with error.

Reply:
Usually I like stack traces, but I see this would be a user-input error through misconfigured .yml, so this should be a message through stderr and an exit code as @mbolivar says.

Internal system error -> stack trace.
User-input error -> error message.

Reply:
fixed